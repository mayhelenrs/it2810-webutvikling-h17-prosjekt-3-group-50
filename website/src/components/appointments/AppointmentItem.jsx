import React from 'react';
import '../../assets/styles/appointment.css';

export class AppointmentItem extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            description: this.props.description,
            color: this.props.color,
            time: this.props.time,
            date: this.props.date,
            checked: false
        }
    }

    handleClick() {
        this.setState({checked: !this.state.checked})
    }
    componentDidUpdate() {
        this.save();
    }

    componentDidMount() {
        const data = this.load();
        if (data !== null) {
            this.setState(prevState => {
                return {...prevState, data}
            });
        }
        this.save();
    }

    save() {
        localStorage.setItem("Appointment" + this.props.id, JSON.stringify(this.state));
    }

    load() {
        return JSON.parse(localStorage.getItem("Appointment" + this.props.id));
    }

    removeAppointment() {
        this.props.handleRemove(this);
        localStorage.removeItem("Appointment" + this.props.id)
    }
    render() {
        const text = this.state.checked;
        return (
            <div className="tile-wrapper" key={this.props.index}>
                <div className="color-tile" style={{
                    backgroundColor: '' + this.state.color,
                    borderColor: '' + this.state.color
                }}></div>
                <div className="textComponents">
                    <div className="titleText">
                        <p>{this.state.description}</p>
                    </div>
                    <div className="timeText">
                        <p>{this.state.time}</p>
                    </div>
                    <div className="dateText">
                        <p>{this.state.date}</p>
                    </div>
                </div>
                <div className="deleteButtonWrap">
                    <input type="button" className="deleteButton" label={text}
                        onClick={() => {this.removeAppointment()}} onChange={() => this.handleClick()}/>
                </div>
            </div>
        );
    }
}
