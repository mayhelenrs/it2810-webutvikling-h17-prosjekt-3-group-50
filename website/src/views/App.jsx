import React from 'react';
import {Switch, Route} from 'react-router-dom'
import '../assets/styles/App.css';
import {Navbar} from '../components';
import {Schedule,AppointmentView, Frontpage, NoteView, ToDoView} from '../views';


export class App extends React.Component {
    render() {
        return (
            <div className="body-fullwidth">
                <div id="content-container">
                    <Switch>
                        <Route exact path='/' component={Frontpage}/>
                        <Route exact path='/todo' component={ToDoView}/>
                        <Route exact path='/schedule' component={Schedule}/>
                        <Route exact path='/appointments' component={AppointmentView}/>
                        <Route exact path='/note' component={NoteView}/>
                    </Switch>
                </div>
            </div>
        );
    }
}
