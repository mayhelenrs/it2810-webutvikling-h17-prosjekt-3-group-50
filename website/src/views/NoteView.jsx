import React from 'react';
import '../assets/styles/NoteView.css';
import {NoteContainer} from '../components/notes/NoteContainer.jsx'
import {Categories} from "../components/categories/Categories";

export class NoteView extends React.Component {

    constructor(props) {
        super(props);
        this.state = {
            noteContainer: <NoteContainer ref={instance => {
                this.noteContainer = instance
            }}/>
        }
    }

    render() {
        return (
            <Categories id={1} title={"notes"} filter={() => this.noteContainer.filter()}>
                {this.state.noteContainer}
            </Categories>
        );
    }

}
