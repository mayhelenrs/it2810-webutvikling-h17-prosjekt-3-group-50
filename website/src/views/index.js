export * from './frontpage.jsx';
export * from './App.jsx';
export * from './AppointmentView.jsx';
export * from './NoteView.jsx';
export * from './Schedule.jsx';
export * from './ToDoView.jsx';
