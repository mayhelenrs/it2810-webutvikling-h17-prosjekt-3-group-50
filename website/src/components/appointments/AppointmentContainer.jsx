import React from 'react';
import update from 'react-addons-update';
import {AppointmentItem} from './AppointmentItem.jsx';
import '../../assets/styles/Appointment.css';
import {LocalStorage} from "../../service/LocalStorage";

export class AppointmentContainer extends React.Component {
    //initiating states and binds functions to this
    constructor(props) {
        super(props);
        this.state = {
            list: [],
            displayList: [],
            desc: '',
            time: '',
            date: ''
        };
        this.appointmentCount = 0;
        this.handleRemove = this.removeAppointment.bind(this);
        this.formSubmit = this.formSubmit.bind(this);
        this.filter = this.filter.bind(this);
    }

    //changes a state when given a target, is used on inputfields
    handleChange(event) {
        this.setState({
            [event.target.name]: event.target.value
        })
    }

    filter() {
        const appointList = this.props.selectedColor() === undefined
            ? this.state.list.map((item) => item)
            : this.state.list.filter((item) => item.props.color === this.props.selectedColor());
        this.setState((prevState) => {
            return {
                ...prevState,
                displayList: appointList
            }
        })
    }

    //retruns selectedColor or a predefined color in none is chosen
    getColor() {
        const color = this.props.selectedColor();
        return color === undefined
            ? '#016D91'
            : color
    }

    //checks if all inputfields have content, then makes a new list so we doesnt directly manipulate state
    //next a new appointment is made and pushed to the new list, then the state gets updated to the new list
    //and inputfields are clear by setting state to nothing
    formSubmit(e) {
        if (this.state.desc.length > 0 && this.state.time.length > 0 && this.state.date.length > 0) {
            const newList = this.state.list.slice();
            const index = newList.length;
            const newAppointment = this.generateAppointmentWithId(this.state.desc, this.state.time, this.state.date, this.getColor(), index);
            newList.push(newAppointment);
            this.setState({
                list: newList,
                desc: '',
                time: '',
                date: ''
            }, () => this.filter())

        }
        e.preventDefault()
    }

    //way to generate and appointment when you dont have any id
    generateAppointment(desc, time, date, color) {
        const id = this.appointmentCount++;
        return this.generateAppointmentWithId(desc, time, date, color, id);
    }

    //simple way to create a new appointment
    generateAppointmentWithId(desc, time, date, color, id) {
        return <AppointmentItem description={desc} time={time} date={date} key={id} color={color} id={id}
                                handleRemove={this.handleRemove}/>
    }

    //when a component is updated its state will be saved
    componentDidUpdate() {
        LocalStorage.save(this.getSaveName(), this.state.list.map(appointment => {
            return {
                desc: appointment.props.description,
                time: appointment.props.time,
                date: appointment.props.date,
                color: appointment.props.color,
                id: appointment.props.id
            };
        }));
    }


    componentDidMount() {
        LocalStorage.load(this.getSaveName(), (data) => {
            let appointments = [];
            data.forEach((appointment) => {
                appointments.push(this.generateAppointmentWithId(appointment.desc, appointment.time, appointment.date, appointment.color, appointment.id));
            });
            this.setState((prevState) => {
                return {...prevState, list: appointments};
            }, () => this.filter());
        });
    }

    //used for deleting one specified appointment by checking if the specified appointment is in the list.
    //if it is, it is remeved and the list is updated to the new list without the specified appointment
    removeAppointment(element) {
        this.state.list.forEach((appointment, index) => {
            if (appointment.props.id === element.props.id) {
                this.setState(prevState => {
                    return {
                        list: update(prevState.list, {
                            $splice: [
                                [index, 1]
                            ]
                        })
                    };
                }, () => this.filter());
            }
        });
    }

    getSaveName() {
        return "AppointmentIds";
    }

    render() {
        return (
            <div>
                <div className="appointmentCards">
                    {this.state.displayList}
                </div>
                <form onSubmit={this.formSubmit} className="formAppointment">
                    <input type="text" className="appointmentInput" name="desc" value={this.state.desc}
                           onChange={this.handleChange.bind(this)} placeholder="Insert description"/>
                    <input type="text" className="appointmentInput" name="time" value={this.state.time}
                           onChange={this.handleChange.bind(this)} placeholder="Insert time"/>
                    <input type="text" className="appointmentInput" name="date" value={this.state.date}
                           onChange={this.handleChange.bind(this)} placeholder="Insert date"/>
                    <button onSubmit={this.formSubmit} id="formButton">
                        Add
                    </button>
                </form>
            </div>
        );
    }
}
