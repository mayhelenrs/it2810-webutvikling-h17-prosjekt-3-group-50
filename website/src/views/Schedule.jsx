import React from 'react';
import {DayGrid} from "../components/schedule/DayGrid";
import {Pagetext} from "../components/Pagetext";
import "../assets/styles/App.css"
import "../assets/styles/Schedule.css"


export class Schedule extends React.Component {

    // Rendering the view, for the under page Schedule.
    render() {
        return(
            <div className={"body-fullwidth"}>
                <div className={"calendar-body"}>
                    <br/>
                    <Pagetext text={"Here is your week schedule!"}/>
                    <br/><br/>
                    <DayGrid/>
                </div>
            </div>

        )
    }

}