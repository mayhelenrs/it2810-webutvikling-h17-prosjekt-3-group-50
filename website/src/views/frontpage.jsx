import React from 'react';
import '../assets/styles/App.css';
import {Pagetext, TileGrid} from '../components';
import {Navbar} from "../components/Navbar";

export class Frontpage extends React.Component {

    render() {
        return (
            <div className="body-fullwidth">
                <Navbar navbarLocation={"DASHBOARD"} navbarImage={require("../assets/images/navbar_lefticon.png")}/>
                <div className="frontpage-body">
                    <br/>
                    <Pagetext text={"VELKOMMEN!"}/>
                    <br/><br/>
                    <div className={"tile-grid-wrapper"}>
                        <TileGrid/>
                    </div>
                </div>
            </div>
        );
    }
}
