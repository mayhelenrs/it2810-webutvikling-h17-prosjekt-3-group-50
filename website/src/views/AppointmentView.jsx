import React from 'react';
import '../assets/styles/NoteView.css';
import {AppointmentContainer} from '../components/appointments/AppointmentContainer.jsx'
import {Categories} from "../components/categories/Categories";
import {Navbar} from "../components/Navbar";

export class AppointmentView extends React.Component {

    constructor(props) {
        super(props);
        this.state = {
            appointmentContainer: <AppointmentContainer ref={instance => {
                this.appointmentContainer = instance
            }}/>
        }
    }

    render() {
        return (
            <div>
                <Navbar navbarLocation={"APPOINTMENTS"} navbarImage={require("../assets/images/navbar_leftarrow.png")}/>
                <Categories id={3} title={"appointments"} filter={() => this.appointmentContainer.filter()}>
                    {this.state.appointmentContainer}
                </Categories>
            </div>
        );
    }

}
