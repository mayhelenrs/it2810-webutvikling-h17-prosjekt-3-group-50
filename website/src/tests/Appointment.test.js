//localStorage is not imported to be used as a class, but to be initiated with jest.
import localStorage from './localStorageMock';

import React from 'react';
import renderer from 'react-test-renderer';
import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import {shallow, mount, render} from 'enzyme';
import {AppointmentView} from "../views/AppointmentView";
import {AppointmentContainer} from "../components/appointments/AppointmentContainer";
import {AppointmentItem} from "../components/appointments/appointmentItem";

Enzyme.configure({ adapter: new Adapter() });




/* Main function for testing the: */
/* Appointment application */
describe('Appointment Overview', () => {
    it('renders the whole Appointment application', () => {
        const instance = renderer.create(<AppointmentView />);
        const tree = instance.toJSON();
        expect(tree).toMatchSnapshot();
    });
});

/* Test function for testing the rendering of: */
/* appointment components */
describe('Appointment Component Renders', () => {
    it('renders the Appointment Component', () => {
        const instance = renderer.create(<AppointmentContainer />);
        const tree = instance.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders the AppointmentItem Component', () => {
        const instance = renderer.create(<AppointmentItem/>);
        const tree = instance.toJSON();
        expect(tree).toMatchSnapshot();
    });
});

// Test instatiation of AppointmentItem and its properties/State.
// Shallow makes sure we only go one level deep and do not care about potential child component behaviour
describe('AppointmentItem', () => {
    const appointment = shallow(<AppointmentContainer/>);
    const wrapper = shallow(
        <AppointmentItem className="items"
                         value={appointment}
                         color={"rgb(43, 186, 178)"}
                         description={"FFS"}
                         date={"24.04.2017"}
                         time={"00:24"}
                         onClick={() => this.handleClicks(index)}/>
    );

    it('instantiates properties and state properly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    //Tests if the handleClick function changes the state from false to true and receives the correct state
    it('handleClick changes state', () => {
        const instance = wrapper.instance();
        expect(instance.state).toEqual({
            "checked": false,
            "color": "rgb(43, 186, 178)",
            "description": "FFS",
            "time": "00:24",
            "date": "24.04.2017"
        });
        instance.handleClick();
        expect(instance.state).toEqual({
            "checked": true,
            "color": "rgb(43, 186, 178)",
            "description": "FFS",
            "time": "00:24",
            "date": "24.04.2017"
        });
    });
    //Creates list for putting in several elements
    it('renderAppointmentItems by data', () => {
        const liste = [
            ["test1", "00:24", "24.04.2017",],
            ["test2", "0:25", "25.04.2017",],
            ["test3", "00:26", "26.04.2017"]
        ];
        const colors = ["rgb(43, 186, 178)", "rgb(43, 186, 178)", "rgb(43, 186, 178)"];

        const instance = appointment.instance();

        instance.setState({list: liste});
        instance.setState({displayList: liste[0]});
        let items = instance.list;
        expect(items).toEqual(instance.displayList);
    });

    it('handleSubmit handles new input from onClick', () => {
        const instance = appointment.instance();
        const liste =[
            ["test1", "00:24", "24.04.2017",],
            ["test2", "0:25", "25.04.2017",],
            ["test3", "00:26", "26.04.2017"]];

        expect(appointment.find('input').length).toEqual(3);
        expect(appointment.find('input').at(1).length).toEqual(1);
        instance.setState({
            list: liste,
            desc: "test4",
            time: "00:30",
            date: "19.11.1999",

        });
        //Checking if state is set
        expect(instance.state.list[0]).toEqual(liste[0]);
        expect(instance.state.desc).toEqual("test4");
        expect(instance.state.time).toEqual("00:30");
        expect(instance.state.date).toEqual("19.11.1999");

    });
});