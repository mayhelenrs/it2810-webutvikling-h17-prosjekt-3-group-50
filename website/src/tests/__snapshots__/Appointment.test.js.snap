// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Appointment Component Renders renders the Appointment Component 1`] = `
<div>
  <div
    className="appointmentCards"
  />
  <form
    className="formAppointment"
    onSubmit={[Function]}
  >
    <input
      className="appointmentInput"
      name="desc"
      onChange={[Function]}
      placeholder="Insert description"
      type="text"
      value=""
    />
    <input
      className="appointmentInput"
      name="time"
      onChange={[Function]}
      placeholder="Insert time"
      type="text"
      value=""
    />
    <input
      className="appointmentInput"
      name="date"
      onChange={[Function]}
      placeholder="Insert date"
      type="text"
      value=""
    />
    <button
      id="formButton"
      onSubmit={[Function]}
    >
      Add
    </button>
  </form>
</div>
`;

exports[`Appointment Component Renders renders the AppointmentItem Component 1`] = `
<div
  className="tile-wrapper"
>
  <div
    className="color-tile"
    style={
      Object {
        "backgroundColor": "undefined",
        "borderColor": "undefined",
      }
    }
  />
  <div
    className="textComponents"
  >
    <div
      className="titleText"
    >
      <p />
    </div>
    <div
      className="timeText"
    >
      <p />
    </div>
    <div
      className="dateText"
    >
      <p />
    </div>
  </div>
  <div
    className="deleteButtonWrap"
  >
    <input
      className="deleteButton"
      label={false}
      onChange={[Function]}
      onClick={[Function]}
      type="button"
    />
  </div>
</div>
`;

exports[`Appointment Overview renders the whole Appointment application 1`] = `
<div
  className="NoteView"
>
  <div
    className="Left-Page"
  >
    <p>
      This is your TODO for now
    </p>
    <div>
      <div
        className="NoteHolder"
      >
        <p>
          <orange>
            Categories
          </orange>
           - Select to filter your notes and add new ones!
        </p>
        <div
          className="Notes"
        >
          <div
            className="CategoryFilter"
            onClick={[Function]}
            style={
              Object {
                "backgroundColor": "#016D91",
                "border": "none",
              }
            }
          />
          <div
            className="CategoryFilter"
            onClick={[Function]}
            style={
              Object {
                "backgroundColor": "#E53F6F",
                "border": "none",
              }
            }
          />
        </div>
      </div>
    </div>
    <div
      className="Middle-Page"
    >
      <div>
        <div
          className="appointmentCards"
        />
        <form
          className="formAppointment"
          onSubmit={[Function]}
        >
          <input
            className="appointmentInput"
            name="desc"
            onChange={[Function]}
            placeholder="Insert description"
            type="text"
            value=""
          />
          <input
            className="appointmentInput"
            name="time"
            onChange={[Function]}
            placeholder="Insert time"
            type="text"
            value=""
          />
          <input
            className="appointmentInput"
            name="date"
            onChange={[Function]}
            placeholder="Insert date"
            type="text"
            value=""
          />
          <button
            id="formButton"
            onSubmit={[Function]}
          >
            Add
          </button>
        </form>
      </div>
    </div>
  </div>
  <div
    className="Right-Page"
  >
    <div
      className="CategoryCreatorHolder"
    >
      <p>
        Categories
      </p>
      <div
        className="Category"
      >
        <div
          className="CategoryColor"
          style={
            Object {
              "backgroundColor": "#016D91",
            }
          }
        />
        <div>
          <input
            className="CategoryName"
            onChange={[Function]}
            value="Other"
          />
        </div>
      </div>
      <div
        className="Category"
      >
        <div
          className="CategoryColor"
          style={
            Object {
              "backgroundColor": "#E53F6F",
            }
          }
        />
        <div>
          <input
            className="CategoryName"
            onChange={[Function]}
            value="School"
          />
        </div>
      </div>
      <div
        className="AddCategory"
      >
        <button
          className="add-button"
          onClick={[Function]}
        >
          Add
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`AppointmentItem instantiates properties and state properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AppointmentItem
    className="items"
    color="rgb(43, 186, 178)"
    date="24.04.2017"
    description="FFS"
    onClick={[Function]}
    time="00:24"
    value={
        ShallowWrapper {
            "length": 1,
            Symbol(enzyme.__root__): [Circular],
            Symbol(enzyme.__unrendered__): <AppointmentContainer />,
            Symbol(enzyme.__renderer__): Object {
              "batchedUpdates": [Function],
              "getNode": [Function],
              "render": [Function],
              "simulateEvent": [Function],
              "unmount": [Function],
            },
            Symbol(enzyme.__node__): Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div
                    className="appointmentCards"
        >
                    
        </div>,
                  <form
                    className="formAppointment"
                    onSubmit={[Function]}
        >
                    <input
                                className="appointmentInput"
                                name="desc"
                                onChange={[Function]}
                                placeholder="Insert description"
                                type="text"
                                value=""
                    />
                    <input
                                className="appointmentInput"
                                name="time"
                                onChange={[Function]}
                                placeholder="Insert time"
                                type="text"
                                value=""
                    />
                    <input
                                className="appointmentInput"
                                name="date"
                                onChange={[Function]}
                                placeholder="Insert date"
                                type="text"
                                value=""
                    />
                    <button
                                id="formButton"
                                onSubmit={[Function]}
                    >
                                Add
                    </button>
        </form>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [],
                    "className": "appointmentCards",
                  },
                  "ref": null,
                  "rendered": Array [],
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <input
                        className="appointmentInput"
                        name="desc"
                        onChange={[Function]}
                        placeholder="Insert description"
                        type="text"
                        value=""
        />,
                      <input
                        className="appointmentInput"
                        name="time"
                        onChange={[Function]}
                        placeholder="Insert time"
                        type="text"
                        value=""
        />,
                      <input
                        className="appointmentInput"
                        name="date"
                        onChange={[Function]}
                        placeholder="Insert date"
                        type="text"
                        value=""
        />,
                      <button
                        id="formButton"
                        onSubmit={[Function]}
        >
                        Add
        </button>,
                    ],
                    "className": "formAppointment",
                    "onSubmit": [Function],
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": null,
                      "nodeType": "host",
                      "props": Object {
                        "className": "appointmentInput",
                        "name": "desc",
                        "onChange": [Function],
                        "placeholder": "Insert description",
                        "type": "text",
                        "value": "",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "input",
                    },
                    Object {
                      "instance": null,
                      "key": null,
                      "nodeType": "host",
                      "props": Object {
                        "className": "appointmentInput",
                        "name": "time",
                        "onChange": [Function],
                        "placeholder": "Insert time",
                        "type": "text",
                        "value": "",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "input",
                    },
                    Object {
                      "instance": null,
                      "key": null,
                      "nodeType": "host",
                      "props": Object {
                        "className": "appointmentInput",
                        "name": "date",
                        "onChange": [Function],
                        "placeholder": "Insert date",
                        "type": "text",
                        "value": "",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "input",
                    },
                    Object {
                      "instance": null,
                      "key": null,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Add",
                        "id": "formButton",
                        "onSubmit": [Function],
                      },
                      "ref": null,
                      "rendered": "Add",
                      "type": "button",
                    },
                  ],
                  "type": "form",
                },
              ],
              "type": "div",
            },
            Symbol(enzyme.__nodes__): Array [
              Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <div
                      className="appointmentCards"
        >
                      
        </div>,
                    <form
                      className="formAppointment"
                      onSubmit={[Function]}
        >
                      <input
                                    className="appointmentInput"
                                    name="desc"
                                    onChange={[Function]}
                                    placeholder="Insert description"
                                    type="text"
                                    value=""
                      />
                      <input
                                    className="appointmentInput"
                                    name="time"
                                    onChange={[Function]}
                                    placeholder="Insert time"
                                    type="text"
                                    value=""
                      />
                      <input
                                    className="appointmentInput"
                                    name="date"
                                    onChange={[Function]}
                                    placeholder="Insert date"
                                    type="text"
                                    value=""
                      />
                      <button
                                    id="formButton"
                                    onSubmit={[Function]}
                      >
                                    Add
                      </button>
        </form>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": null,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [],
                      "className": "appointmentCards",
                    },
                    "ref": null,
                    "rendered": Array [],
                    "type": "div",
                  },
                  Object {
                    "instance": null,
                    "key": null,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        <input
                          className="appointmentInput"
                          name="desc"
                          onChange={[Function]}
                          placeholder="Insert description"
                          type="text"
                          value=""
        />,
                        <input
                          className="appointmentInput"
                          name="time"
                          onChange={[Function]}
                          placeholder="Insert time"
                          type="text"
                          value=""
        />,
                        <input
                          className="appointmentInput"
                          name="date"
                          onChange={[Function]}
                          placeholder="Insert date"
                          type="text"
                          value=""
        />,
                        <button
                          id="formButton"
                          onSubmit={[Function]}
        >
                          Add
        </button>,
                      ],
                      "className": "formAppointment",
                      "onSubmit": [Function],
                    },
                    "ref": null,
                    "rendered": Array [
                      Object {
                        "instance": null,
                        "key": null,
                        "nodeType": "host",
                        "props": Object {
                          "className": "appointmentInput",
                          "name": "desc",
                          "onChange": [Function],
                          "placeholder": "Insert description",
                          "type": "text",
                          "value": "",
                        },
                        "ref": null,
                        "rendered": null,
                        "type": "input",
                      },
                      Object {
                        "instance": null,
                        "key": null,
                        "nodeType": "host",
                        "props": Object {
                          "className": "appointmentInput",
                          "name": "time",
                          "onChange": [Function],
                          "placeholder": "Insert time",
                          "type": "text",
                          "value": "",
                        },
                        "ref": null,
                        "rendered": null,
                        "type": "input",
                      },
                      Object {
                        "instance": null,
                        "key": null,
                        "nodeType": "host",
                        "props": Object {
                          "className": "appointmentInput",
                          "name": "date",
                          "onChange": [Function],
                          "placeholder": "Insert date",
                          "type": "text",
                          "value": "",
                        },
                        "ref": null,
                        "rendered": null,
                        "type": "input",
                      },
                      Object {
                        "instance": null,
                        "key": null,
                        "nodeType": "host",
                        "props": Object {
                          "children": "Add",
                          "id": "formButton",
                          "onSubmit": [Function],
                        },
                        "ref": null,
                        "rendered": "Add",
                        "type": "button",
                      },
                    ],
                    "type": "form",
                  },
                ],
                "type": "div",
              },
            ],
            Symbol(enzyme.__options__): Object {
              "adapter": ReactSixteenAdapter {
                "options": Object {
                  "enableComponentDidUpdateOnSetState": true,
                },
              },
            },
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="color-tile"
          style={
                    Object {
                              "backgroundColor": "rgb(43, 186, 178)",
                              "borderColor": "rgb(43, 186, 178)",
                            }
          }
/>,
        <div
          className="textComponents"
>
          <div
                    className="titleText"
          >
                    <p>
                              FFS
                    </p>
          </div>
          <div
                    className="timeText"
          >
                    <p>
                              00:24
                    </p>
          </div>
          <div
                    className="dateText"
          >
                    <p>
                              24.04.2017
                    </p>
          </div>
</div>,
        <div
          className="deleteButtonWrap"
>
          <input
                    className="deleteButton"
                    label={false}
                    onChange={[Function]}
                    onClick={[Function]}
                    type="button"
          />
</div>,
      ],
      "className": "tile-wrapper",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "className": "color-tile",
          "style": Object {
            "backgroundColor": "rgb(43, 186, 178)",
            "borderColor": "rgb(43, 186, 178)",
          },
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="titleText"
>
              <p>
                            FFS
              </p>
</div>,
            <div
              className="timeText"
>
              <p>
                            00:24
              </p>
</div>,
            <div
              className="dateText"
>
              <p>
                            24.04.2017
              </p>
</div>,
          ],
          "className": "textComponents",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": <p>
                FFS
</p>,
              "className": "titleText",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": "FFS",
              },
              "ref": null,
              "rendered": "FFS",
              "type": "p",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": <p>
                00:24
</p>,
              "className": "timeText",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": "00:24",
              },
              "ref": null,
              "rendered": "00:24",
              "type": "p",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": <p>
                24.04.2017
</p>,
              "className": "dateText",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": "24.04.2017",
              },
              "ref": null,
              "rendered": "24.04.2017",
              "type": "p",
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": <input
            className="deleteButton"
            label={false}
            onChange={[Function]}
            onClick={[Function]}
            type="button"
/>,
          "className": "deleteButtonWrap",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "className": "deleteButton",
            "label": false,
            "onChange": [Function],
            "onClick": [Function],
            "type": "button",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="color-tile"
            style={
                        Object {
                                    "backgroundColor": "rgb(43, 186, 178)",
                                    "borderColor": "rgb(43, 186, 178)",
                                  }
            }
/>,
          <div
            className="textComponents"
>
            <div
                        className="titleText"
            >
                        <p>
                                    FFS
                        </p>
            </div>
            <div
                        className="timeText"
            >
                        <p>
                                    00:24
                        </p>
            </div>
            <div
                        className="dateText"
            >
                        <p>
                                    24.04.2017
                        </p>
            </div>
</div>,
          <div
            className="deleteButtonWrap"
>
            <input
                        className="deleteButton"
                        label={false}
                        onChange={[Function]}
                        onClick={[Function]}
                        type="button"
            />
</div>,
        ],
        "className": "tile-wrapper",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "className": "color-tile",
            "style": Object {
              "backgroundColor": "rgb(43, 186, 178)",
              "borderColor": "rgb(43, 186, 178)",
            },
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="titleText"
>
                <p>
                                FFS
                </p>
</div>,
              <div
                className="timeText"
>
                <p>
                                00:24
                </p>
</div>,
              <div
                className="dateText"
>
                <p>
                                24.04.2017
                </p>
</div>,
            ],
            "className": "textComponents",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": <p>
                  FFS
</p>,
                "className": "titleText",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "children": "FFS",
                },
                "ref": null,
                "rendered": "FFS",
                "type": "p",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": <p>
                  00:24
</p>,
                "className": "timeText",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "children": "00:24",
                },
                "ref": null,
                "rendered": "00:24",
                "type": "p",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": <p>
                  24.04.2017
</p>,
                "className": "dateText",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "children": "24.04.2017",
                },
                "ref": null,
                "rendered": "24.04.2017",
                "type": "p",
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": <input
              className="deleteButton"
              label={false}
              onChange={[Function]}
              onClick={[Function]}
              type="button"
/>,
            "className": "deleteButtonWrap",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "className": "deleteButton",
              "label": false,
              "onChange": [Function],
              "onClick": [Function],
              "type": "button",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
